File: medium/Linked List/Rotate List.py
class Solution(object):
    def rotateRight(self, head, k):
        if head==None or head.next ==None or k==0:
            return head
        curr = head
        l = 1
        while curr.next:
            curr = curr.next
            l+=1
        r = k%l
        k = l-r
        curr.next = head
        while k>0:
            curr = curr.next
            k-=1
        head=curr.next
        curr.next = None
        return head
        

File: medium/Linked List/Rotate List.cpp
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        if(!head || !head->next  || k ==0 ) return head;
        int l = 1;
        ListNode* curr = head;
        while(curr->next){
            l++; curr = curr->next;
        }k = k%l;
        k = l-k;
        curr->next = head;
        while(k-- > 0) curr = curr->next;
        head = curr->next;
        curr->next = nullptr;
        return head;
    }
};